app/controllers/course/announcements_controller.rb

```diff
@@ -4,6 +4,10 @@ class Course::AnnouncementsController < Course::ModuleController
   def index #:nodoc:
     @announcements = @announcements.includes(:creator).sorted_by_sticky.sorted_by_date
     @announcements = @announcements.page(params[:page])
+    # mark each announcement in this page as read
+    @announcements.each do |ann|
```

Comment:
Maybe not. unread_by() is a class method. Maybe there is no way to get unread announcements immediately from @announcements. Also there is a function can mark every announcements in every courses as read. That is not what we want...